addons/mail/static/src/models/discuss/discuss.js

```diff
@@ -160,11 +150,32 @@ function factory(dependencies) {
             if (!thread) {
                 return;
             }
+            thread.open();
+        }
+
+
+        /**
+         * Opens the given thread in Discuss, and opens Discuss if necessary.
+         *
+         * @param {mail.thread} thread
+         */
+        openThread(thread) {
             this.threadViewer.update({
                 stringifiedDomain: '[]',
                 thread: [['link', thread]],
             });
-            thread.open({ resetDiscussDomain: true });
+            // FIXME this focus is too early if Discuss is not yet open
+            this.threadViewer.composer.focus();
+            if (!this.isOpen) {
+                this.env.bus.trigger('do-action', {
+                    action: 'mail.action_discuss',
+                    options: {
```

Comment:
can we get a callback from when the action is done? to do focus and properly await